lib/ansible/modules/mount_facts.py

```diff
@@ -0,0 +1,472 @@
+        if not source:
+            module.fail_json(msg="sources contains an empty string")
+
+        if source in {"static", "all"}:
+            sources.extend(STATIC_SOURCES)
+        if source in {"dynamic", "all"}:
+            sources.extend(DYNAMIC_SOURCES)
+
+        elif source not in {"static", "dynamic", "all", module.params["mount_binary"]}:
+            sources.append(source)
+    return sources
+
+
+def gen_mounts_by_source(module: AnsibleModule):
+    """Iterate over the sources and yield tuples containing the source, mount point, source line(s), and the parsed result."""
+    sources = get_file_sources(module)
+
+    if len(set(sources)) < len(sources):
+        module.warn(f"mount_facts option 'sources' contains duplicate entries, repeat sources will be ignored: {sources}")
+
+    collected = set()
+    for mount_tuple in gen_mounts_by_file(sources):
+        collected.add(mount_tuple[0])
+        yield mount_tuple
+
+    mount_binary = module.params["mount_binary"]
+    explicit_mount_source = mount_binary and mount_binary in (module.params["sources"] or [])
+    implicit_mount_source = mount_binary and set(sources).intersection(DYNAMIC_SOURCES) and not collected.intersection(DYNAMIC_SOURCES)
+    if explicit_mount_source or implicit_mount_source:
+        stdout = run_mount_bin(module, mount_binary)
+        yield from [(mount_binary, *mount_info) for mount_info in gen_mounts_from_stdout(stdout)]
+
+
+def get_mount_facts(module: AnsibleModule, uuids: dict, udevadm_uuid: Callable):
+    """List and filter mounts, returning a dictionary containing mount points as the keys (last listed wins)."""
+    seconds = module.params["timeout"]
+
+    # merge sources based on the mount point (last source wins)
+    facts = {}
+    for source, mount, origin, fields in gen_mounts_by_source(module):
+        device = fields["device"]
+        fstype = fields["fstype"]
+
+        if not any(fnmatch(device, pattern) for pattern in module.params["devices"] or ["*"]):
+            continue
+        if not any(fnmatch(fstype, pattern) for pattern in module.params["fstypes"] or ["*"]):
+            continue
+
+        timed_func = _timeout.timeout(seconds, f"Timed out getting mount size for mount {mount} (type {fstype})")(get_mount_size)
+        if mount_size := handle_timeout(module)(timed_func)(mount):
+            fields.update(mount_size)
+
+        timed_func = _timeout.timeout(seconds, f"Timed out getting uuid for mount {mount} (type {fstype})")(udevadm_uuid)
+        uuid = uuids.get(device, handle_timeout(module)(timed_func)(device))
+
+        fields.update({"ansible_context": {"source": source, "source_data": origin}, "uuid": uuid or "N/A"})
+        facts[mount] = fields
+
+    return facts
+
+
+def get_argument_spec():
+    """Helper returning the argument spec."""
+    return dict(
+        sources=dict(type="list", elements="str", default=None),
+        mount_binary=dict(default="mount", type="raw"),
```

Comment:
Hopefully I've captured this pretty well in the documentation, but wanted to add a note on the implementation, since type `raw` is out of the ordinary. This gets used implicitly as a dynamic source only for hosts that don't have a file-based alternative, which is why it's configurable alongside sources. This is type `raw` to allow `mount_binary: ~` (to opt out of the fallback) since `~` is not a valid choice for a str/path type unless it is the default value. The default isn't `~` to allow BSD and AIX to collect the current mounts by default.

The mount_binary can also be configured explicitly in sources so it always runs, but this would be redundant for Linux (only really makes sense as the only dynamic source in that case, but the mount output is non-standardized, and parsing it is best-effort, using a file-based source is superior).